imap_processing/codice/l0/decompress_codice.py

```diff
@@ -0,0 +1,111 @@
+"""Decompress CoDICE science data.
+
+For CoDICE, there are 3 forms of compression:
+    1. Table-based lossy compression A (24 -> 8-bit)
+    2. Table-based lossy compression B (24 -> 8 bit)
+    3. LZMA lossless compression
+
+Only one lossy option can be selected in cases of lossy + lossless compression.
+Thus, there are 6 possibly compression algorithms:
+    1. No compression
+    2. Lossy A only
+    3. Lossy B only
+    4. Lossless only
+    5. Lossy A + lossless
+    6. Lossy B + lossless
+
+In the case of (5) and (6), the data is first run through lossy compression, and
+then the result is run through lossless compression. Thus, to decompress, one
+must apply lossless decompression first, then lossy decompression
+
+References
+----------
+    This information was provided via email from Greg Dunn on Oct 23, 2023
+"""
+
+import lzma
+
+from . import LOSSY_A_TABLE, LOSSY_B_TABLE
+
+
+class CodiceDecompression:
```

Comment:
Is there a reason to use a class here? This seems like a function to me: `def decompress_codice(compressed_value, algorithm) -> decompressed_value`
Basically keeping all the functions you have and just returning instead of setting attributes on self.

Reply:
Yeah I went back and forth on this. I don't really have a good reason here to use a class, but now that I reworked this into using functions, I do think the code is a bit cleaner. So good suggestion here.